ui-ngx/src/app/core/services/utils.service.ts

```diff
@@ -374,6 +381,39 @@ export class UtilsService {
     });
   }
 
+  public getUserDisplayName(alarmAssignee: AlarmAssignee |  AlarmCommentInfo) {
+    let displayName = '';
+    if (isNotEmptyStr(alarmAssignee.firstName) || isNotEmptyStr(alarmAssignee.lastName)) {
+      if (alarmAssignee.firstName) {
+        displayName += alarmAssignee.firstName;
+      }
+      if (alarmAssignee.lastName) {
+        if (displayName.length > 0) {
+          displayName += ' ';
+        }
+        displayName += alarmAssignee.lastName;
+      }
+    } else {
+      displayName = alarmAssignee.email;
+    }
+    return displayName;
+  }
+
+  getUserInitials(alarmAssignee: AlarmAssignee): string {
+    let initials = '';
+    if (isNotEmptyStr(alarmAssignee.firstName) || isNotEmptyStr(alarmAssignee.lastName)) {
+      if (alarmAssignee.firstName) {
+        initials += alarmAssignee.firstName.charAt(0);
+      }
+      if (alarmAssignee.lastName) {
+        initials += alarmAssignee.lastName.charAt(0);
+      }
+    } else {
+      initials += alarmAssignee.email.charAt(0);
+    }
+    return initials.toUpperCase();
+  }
```

Comment:
Maybe it's better to move these functions in a file to AlarmAssignee/AlarmCommentInfo model?